RELEASE.md

```diff
@@ -0,0 +1,304 @@
+# Release process
+
+The release process starts with one or more tagged release candidates,
+when no more blocking issues needs to be fixed a final tag is created
+and the final release is rolled.
```

Comment:
I'm guessing you are copying this wording from somewhere else... but it doesn't read that well to me. Here's an improvement (i think). Not perfect, but it'll do.

"The release process starts by tagging and testing one ore more release candidates. After a period of time has passed and there are no remaining blocking issues, the final release tag is created and release rolled."